### PR TITLE
 Non-blocking send, additional receive path fixes, purge send minimization [DEVC-841] [v1.6]

### PR DESCRIPTION
**Pull request by silverjam**
Release 1.6 version of #711 

+ Our implementation pk_endpoint_send call nn_send with NN_DONTWAIT so that we don't block on outgoing packets

+ Fix our receive path by looking for the right error codes, also allow receiver callbacks to terminate the loop

+ Reduce the number of packets we'll service in one loop to 32

+ Bump to nanomsg 1.1.4 instead of 1.0.0 (which is the latest in buildroot master)

+ Remove send minimization since it doesn't seem to help now that we're fixed the blocking send call

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.6.8-7-gd504c8f3
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=580942)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.6.8-7-gd504c8f3&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.6.8-7-gd504c8f3&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.6.8-7-gd504c8f3
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.6.8-7-gd504c8f3)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.6.8-7-gd504c8f3/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.6.8-7-gd504c8f3

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.6.8-7-gd504c8f3
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=580955)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.6.8-7-gd504c8f3&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.6.8-7-gd504c8f3&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.6.8-7-gd504c8f3
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.6.8-7-gd504c8f3)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.6.8-7-gd504c8f3/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.6.8-7-gd504c8f3

File: package/endpoint_adapter/src/endpoint_adapter.c [hunk 1]
```diff
@@ -91,8 +91,6 @@ static const u64 one_second_ns = 1e9;
 static u64 last_metrics_flush = 0;
 static u64 last_send_flush = 0;
```
Comment (benjaminaltieri) [inline package/endpoint_adapter/src/endpoint_adapter.c]: this looks unused

File: package/libpiksi/libpiksi/src/sbp_pubsub.c [hunk 2]
```diff
@@ -15,8 +15,6 @@
 #include <libpiksi/sbp_pubsub.h>
 
 struct sbp_pubsub_ctx_s {
-  pk_endpoint_t *pub_ept;
```
Comment (benjaminaltieri) [inline package/libpiksi/libpiksi/src/sbp_pubsub.c]: 👍

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.6.8-8-g26cf8201
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=581054)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.6.8-8-g26cf8201&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.6.8-8-g26cf8201&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.6.8-8-g26cf8201
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.6.8-8-g26cf8201)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.6.8-8-g26cf8201/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.6.8-8-g26cf8201

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v1.6.8-8-g26cf8201
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=581055)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.6.8-8-g26cf8201&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v1.6.8-8-g26cf8201&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v1.6.8-8-g26cf8201
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v1.6.8-8-g26cf8201)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.6.8-8-g26cf8201/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.6.8-8-g26cf8201

**Pull request merged**